torch/csrc/distributed/rpc/process_group_agent.cpp

```diff
@@ -147,10 +147,14 @@ ProcessGroupAgent::ProcessGroupAgent(
   for (int rank = 0; rank < (int)tmpWorkerIds.size(); ++rank) {
     allWorkerInfo_.emplace_back(std::move(tmpWorkerIds[rank]), rank);
   }
+
+  // construct PythonRpcHandler singleton here
+  PythonRpcHandler::getInstance();
```

Comment:
@rohan-varma why is this needed? I removed it because I tried to make process group agent independent of python. PythonRpcHandler has dependency on  python interpreter. if process group agent is used in pure C++, this line will not work.